drivers/bus/isapnp/hardware.c

```diff
@@ -661,28 +634,19 @@ ParseTags(
             {
                 PISAPNP_IRQ_DESCRIPTION Description;
 
-                if (LogDev != 0 ||
-                    (TagLen > sizeof(ISAPNP_IRQ_DESCRIPTION) ||
-                     TagLen < (sizeof(ISAPNP_IRQ_DESCRIPTION) - 1)) ||
-                    NumberOfIrq >= RTL_NUMBER_OF(LogDevice->Irq))
-                {
+                if (LogDev != 0)
                     goto SkipTag;
-                }
 
-                if (DfState == dfStarted)
```

Comment:
Can you explain a little bit (not in the code, but somewhere in the PR) why it is actually not needed to look at that "DfState" thing anymore for validating/retrieving that information anymore?
Maybe you could tell us what you think was the original "idea" why the old code did that (but is not needed now)?

Reply:
This validation is already done by isapnp!IsaHwConfigureDevice, and furthermore the implementation here was incorrect for any card with a complex resource data structure with a lot of dependent-function resource descriptors.